src/presign/participant.rs

```diff
@@ -75,21 +103,31 @@ impl PresignParticipant {
     ) -> Result<(Option<PresignRecord>, Vec<Message>)> {
         match message.message_type() {
             MessageType::Presign(PresignMessageType::Ready) => {
-                let messages = self.handle_ready_msg(rng, message, main_storage)?;
-                Ok((None, messages))
+                let result = self.handle_ready_msg(rng, message, main_storage)?;
+                Ok(result)
+            }
+            MessageType::Presign(PresignMessageType::RoundOneBroadcast) => {
+                let (broadcast_option, mut messages) = self.handle_broadcast(rng, message)?;
+                if let Some(bmsg) = broadcast_option {
+                    let (pr, more_messages) =
+                        self.handle_round_one_broadcast_msg(rng, &bmsg, main_storage)?;
+                    messages.extend_from_slice(&more_messages);
+                    Ok((pr, messages))
+                } else {
+                    Ok((None, messages))
+                }
             }
             MessageType::Presign(PresignMessageType::RoundOne) => {
-                let messages = self.handle_round_one_msg(rng, message, main_storage)?;
-                Ok((None, messages))
+                let result = self.handle_round_one_msg(rng, message, main_storage)?;
+                Ok(result)
             }
             MessageType::Presign(PresignMessageType::RoundTwo) => {
-                let messages = self.handle_round_two_msg(rng, message, main_storage)?;
-                Ok((None, messages))
+                let result = self.handle_round_two_msg(rng, message, main_storage)?;
+                Ok(result)
             }
             MessageType::Presign(PresignMessageType::RoundThree) => {
-                let (presign_record_option, messages) =
-                    self.handle_round_three_msg(rng, message, main_storage)?;
-                Ok((presign_record_option, messages))
+                let result = self.handle_round_three_msg(rng, message, main_storage)?;
+                Ok(result)
```

Comment:
Similar to my last comment, we can omit the intermediate variable in these match arms.